File: frontend/lib/api/types/skill-passport.ts
export type SkillPassportEvidence = {
  id: string;
  title: string | null;
  description: string | null;
  source_type: string;
  source_reference: string | null;
  evidence_confidence: number;
};

export type SkillPassportSkill = {
  id: string;
  name: string;
  category: string;
  evidence_confidence: number;
  evidence_count: number;
  evidence: SkillPassportEvidence[];
  github_repositories: { repository_name: string; repository_url: string; evidence_count: number; repository_confidence: number }[];
};

export type SkillPassportResponse = {
  skills: SkillPassportSkill[];
  total_skills: number;
  total_evidence: number;
};

File: backend/app/schemas/skill_passport.py
from uuid import UUID

from pydantic import BaseModel


class SkillPassportEvidenceResponse(BaseModel):
    id: UUID
    title: str | None
    description: str | None
    source_type: str
    source_reference: str | None
    evidence_confidence: float


class SkillPassportGitHubRepositoryResponse(BaseModel):
    repository_name: str
    repository_url: str
    evidence_count: int
    repository_confidence: int


class SkillPassportSkillResponse(BaseModel):
    id: UUID
    name: str
    category: str
    evidence_confidence: float
    evidence_count: int
    evidence: list[SkillPassportEvidenceResponse]
    github_repositories: list[SkillPassportGitHubRepositoryResponse] = []


class SkillPassportResponse(BaseModel):
    skills: list[SkillPassportSkillResponse]
    total_skills: int
    total_evidence: int
